src/Book.ts

```diff
@@ -0,0 +1,197 @@
+import { Entry } from "./Entry";
+import { IPaginationQuery, IParseQuery, parseQuery } from "./helper/parseQuery";
+import { IJournal, journalModel, TJournalDocument } from "./models/journals";
+import {
+  isValidTransactionKey,
+  ITransaction,
+  transactionModel,
+} from "./models/transactions";
+import type { IOptions } from "./IOptions";
+import type { Document, PipelineStage, Types } from "mongoose";
+
+export class Book<
+  U extends ITransaction = ITransaction,
+  J extends IJournal = IJournal
+> {
+  name: string;
+  precision: number;
+
+  constructor(name: string, options = {} as { precision?: number }) {
+    this.name = name;
+    this.precision =
+      typeof options.precision === "number" ? options.precision : 7;
+  }
+
+  entry(
+    memo: string,
+    date = null as Date | null,
+    original_journal = null as string | Types.ObjectId | null
+  ): Entry<U, J> {
+    return Entry.write<U, J>(this, memo, date, original_journal);
+  }
+
+  async balance(
+    query: IParseQuery & IPaginationQuery,
+    options = {} as IOptions
+  ): Promise<{ balance: number; notes: number }> {
+    let skip: PipelineStage.Skip | undefined = undefined;
+
+    if (query.perPage) {
+      skip = { $skip: (query.page ? query.page - 1 : 0) * query.perPage };
```

Comment:
Balance only uses skip for the pagination but not a limit. This results in the effect, that the balance sums up. E.g. you do a .ledger call with the same parameters and balance and ledger will not sum to the same value. 

A potential fix would be https://github.com/flash-oss/medici/commit/50d928f90bd2b518e51deee75452c1851d939ca9

Maybe pagination in balance makes no sense anyway, and should be removed.

To everybody:
What do you think?